core/frontend/src/DrawingViewState.ts

```diff
@@ -0,0 +1,419 @@
+
+      const branch = context.createGraphicBranch(graphics, this._toDrawing, this._branchOptions);
+      context.outputGraphic(branch);
+    };
+
+    outputGraphics(scene.foreground);
+    context.withGraphicType(TileGraphicType.BackgroundMap, () => outputGraphics(scene.background));
+    context.withGraphicType(TileGraphicType.Overlay, () => outputGraphics(scene.overlay));
+
+    // Report tile statistics to drawing viewport.
+    const tileAdmin = IModelApp.tileAdmin;
+    const selectedAndReady = tileAdmin.getTilesForViewport(this.viewport);
+    const requested = tileAdmin.getRequestsForViewport(this.viewport);
+    tileAdmin.addExternalTilesForViewport(context.viewport, {
+      requested: requested?.size ?? 0,
+      selected: selectedAndReady?.selected.size ?? 0,
+      ready: selectedAndReady?.ready.size ?? 0,
+    });
+  }
+}
+
+const defaultSectionDrawingProps = {
+  spatialView: Id64.invalid,
+  displaySpatialView: false,
+};
+
+/** A view of a [DrawingModel]($backend)
+ * @public
+ */
+export class DrawingViewState extends ViewState2d {
+  /** @internal */
+  public static get className() { return "DrawingViewDefinition"; }
+
+  /** Exposed strictly for testing and debugging. Indicates that when loading the view, the spatial view should be displayed even
+   * if `SectionDrawing.displaySpatialView` is not `true`.
+   * @internal
+   */
+  public static alwaysDisplaySpatialView = false;
+
+  /** Exposed strictly for testing and debugging. Indicates that the 2d graphics should not be displayed.
+   * @internal
+   */
+  public static hideDrawingGraphics = false;
+
+  private readonly _modelLimits: ExtentLimits;
+  private readonly _viewedExtents: AxisAlignedBox3d;
+  private _attachment?: SectionAttachment;
+  private _removeTileLoadListener?: () => void;
+  private _sectionDrawingProps?: SectionDrawingViewProps;
+
+  /** Strictly for testing. @internal */
+  public get sectionDrawingInfo() {
+    return this._attachment?.sectionDrawingInfo;
+  }
+
+  /** Strictly for testing. @internal */
+  public get sectionDrawingProps(): Readonly<SectionDrawingViewProps> | undefined {
+    return this._sectionDrawingProps;
+  }
+
+  public constructor(props: ViewDefinition2dProps, iModel: IModelConnection, categories: CategorySelectorState, displayStyle: DisplayStyle2dState, extents: AxisAlignedBox3d, sectionDrawing?: SectionDrawingViewProps) {
+    super(props, iModel, categories, displayStyle);
+    if (categories instanceof DrawingViewState) {
+      this._viewedExtents = categories._viewedExtents.clone();
+      this._modelLimits = { ...categories._modelLimits };
+      this._attachment = categories._attachment?.clone();
+      this._sectionDrawingProps = categories._sectionDrawingProps;
+    } else {
+      this._viewedExtents = extents;
+      this._modelLimits = { min: Constant.oneMillimeter, max: 10 * extents.maxLength() };
+      this._sectionDrawingProps = sectionDrawing ?? defaultSectionDrawingProps;
```

Comment:
what's the difference between undefined and defaultSectionDrawingProps?

Reply:
`undefined` means we don't know what the section drawing props are. This occurs as a result of `changeViewedModel`.